multishard_mutation_query.cc

```diff
@@ -592,7 +592,8 @@ static future<reconcilable_result> do_query_mutations(
                     tracing::trace_state_ptr trace_state,
                     streamed_mutation::forwarding,
                     mutation_reader::forwarding fwd_mr) {
-                return make_multishard_combining_reader(ctx, dht::global_partitioner(), std::move(s), pr, ps, pc, std::move(trace_state), fwd_mr);
+                    auto& partitioner = s->get_partitioner();
+                return make_multishard_combining_reader(ctx, partitioner, std::move(s), pr, ps, pc, std::move(trace_state), fwd_mr);
```

Comment:
Yes, also having two arguments enables bugs related to passing a schema and a mismatching partitioner.

Reply:
I agree with you both. Fixed. Thank you for catching. Unfortunately it can't be fixed before the ability to add partitioner to the schema is added because some tests use this method with dummy_partitioner. Will fix as a commit on top of the PR